pandas/core/generic.py

```diff
@@ -11143,7 +11143,7 @@ def __ixor__(self, other):
     # Misc methods
 
     @final
-    def _find_valid_index(self, how: str):
+    def _find_valid_index(self, *, how: str) -> Optional[Hashable]:
```

Comment:
just Hashable? xref #40534

Reply:
I commented on that here in in #40534: https://github.com/pandas-dev/pandas/pull/40534#issuecomment-804010959.

I don't have a definite opinion, but this is the same as in many other cases in the code base and emphasizes that None is special, i.e. in `df[df.find_first_valid(): df.find_last_valid()]` a return value in `_find_valid_index` of None means slice to the start/end and not to that value.

If this should be just `Hashable`, we should probably do that in the whole code base.

Reply:
make sense